File: SelfDescribingNumbers/sdn.py
# Self Describing Numbers solution in Python 2 for CodeEval.com by Steven A. Dunn

def isSelfDescribing(n):
  counts = [0] * 10

  for c in n:
    counts[int(c)] += 1

  for i in range(0, len(n)):
    c = int(n[i])
    if not c == counts[i]: return False

  return True;

import sys

for line in open(sys.argv[1], 'r'):
  if isSelfDescribing(line.rstrip('\n')): print 1
  else: print 0

File: SelfDescribingNumbers/Ruby/sdn.rb
# self describing numbers soln in rb for codeeval.com by steven a dunn

def isSelfDescribing(n)
    counts = Array.new(10, 0)

    n.split("").each do |c|
        counts[c.to_i] += 1
    end

    (0..n.length).each do |i|
        c = n[i].to_i
        unless c == counts[i]
            return false
        end
    end

    return true;
end

File.open(ARGV[0], 'r').each_line do |line|
    if isSelfDescribing(line.chomp)
        puts 1
    else
        puts 0
    end
end